packages/api/src/providers/oauth1.ts

```diff
@@ -161,23 +161,26 @@ export class OAuth1DefaultImpl implements OAuth1 {
     return { oauth_token, oauth_token_secret };
   }
 
-  async deregister(userAccessTokens: string[]): Promise<void> {
+  async deregister(userAccessTokens: string[], cxId?: string): Promise<void> {
     for (const oauthUserAccessToken of userAccessTokens) {
       const userTokenList = await getUserTokenByUAT({ oauthUserAccessToken });
       for (const userToken of userTokenList) {
-        capture.setUser({ id: userToken.userId });
         // DynamoDB (Webhook and auth)
-        const updatedUserToken = userToken.clone();
-        updatedUserToken.oauthUserAccessToken = undefined;
-        updatedUserToken.oauthUserAccessSecret = undefined;
-        await saveUserToken(updatedUserToken);
-        // Postgres (app standard)
-        await updateProviderData({
-          id: userToken.userId,
-          cxId: userToken.cxId,
-          provider: this.providerName,
-          providerItem: undefined,
-        });
+        if (!cxId || cxId === userToken.cxId) {
+          capture.setUser({ id: userToken.userId });
+          const updatedUserToken = userToken.clone();
+          updatedUserToken.oauthUserAccessToken = undefined;
+          updatedUserToken.oauthUserAccessSecret = undefined;
+          await saveUserToken(updatedUserToken);
+
+          // Postgres (app standard)
+          await updateProviderData({
+            id: userToken.userId,
+            cxId: userToken.cxId,
+            provider: this.providerName,
+            providerItem: undefined,
+          });
+        }
```

Comment:
Should we wrap the loop contents with a try/catch, so a failure to deregister a provider doesn't prevent us from deregistering with the rest?